### ISSUE TITLE
Using vkCmdClearAttachments with no stencil attachment sometimes causes Metal validation error

### ISSUE THREAD

**Issue by mellinoe** [KhronosGroup/MoltenVK#65]
I'm seeing that calling `vkCmdClearAttachments` under MoltenVK sometimes causes this sort of validation error:

```
validateDepthStencilState:3671: failed assertion `MTLDepthStencilDescriptor uses frontFaceStencil but MTLRenderPassDescriptor has a nil stencilAttachment texture'
```

From the stack trace, it looks like this validation error is being hit in a synthetic draw command being issued inside of `MVKCmdClearAttachments::encode`. The error does not get triggered until the command buffer is actually submitted/executed.

Note that my application doesn't have or attempt to use a stencil texture or attachment.

**Comment by mellinoe** [comment]
It turns out this was caused by me passing in both `VK_IMAGE_ASPECT_DEPTH_BIT` and `VK_IMAGE_ASPECT_STENCIL_BIT` to `vkCmdClearAttachments` when I only had a depth attachment. There's no validation error for this (and it works on all other drivers I have tried), but the spec does seem to say that you shouldn't do this. I'll leave this open just to ensure that I'm reading the spec right and this isn't permitted.
